app/controllers/course/assessment/question/text_responses_controller.rb

```diff
@@ -6,12 +6,16 @@ class Course::Assessment::Question::TextResponsesController < \
                               through: :assessment, parent: false
 
   def new
+    if params[:file_upload] == 'true'
+      @text_response_question.hide_text = true
```

Comment:
Possible - if we rename this question to `Course::Assessment::Question::General` maybe `allow_text_response` would be a good column name. For now I used `hide_text` in relation to the class name.